src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -880,7 +888,7 @@ pub extern "C" fn materialize_directories(
     })
     .collect();
 
-  let dir_and_digests = match directories_paths_and_digests_results {
+  let dir_and_digests: Vec<(PathBuf, Digest)> = match directories_paths_and_digests_results {
```

Comment:
Can the changes to this function be reverted? It looks like this may have been just added while trying to grok. IIUC you'll be killing this function anyhow once all callers can be converted to making a product request.

Reply:
yup that's what happened, will remove